perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2273,6 +2273,31 @@ public Member createSponsoredMember(PerunSession session, Vo vo, String namespac
 		return sponsoredMember;
 	}
 
+	@Override
+	public Map<String, String> createSponsoredMembers(PerunSession sess, Vo vo, String namespace, List<String> names, User sponsor, boolean asyncValidation) throws AttributeNotExistsException, WrongAttributeAssignmentException {
+		Map<String, String> result = new HashMap<>();
+		PasswordManagerModule module = getPerunBl().getUsersManagerBl().getPasswordManagerModule(sess, namespace);
+
+		for (String name : names) {
+			// generate random password
+			String password = module.generateRandomPassword(sess, null);
+			Map<String, String> mapName = new HashMap<>();
+			mapName.put("guestName", name);
+			// create sponsored member
+			RichMember richMember;
+			try {
+				richMember = getRichMember(sess, createSponsoredMember(sess, vo, namespace, mapName, password, sponsor, asyncValidation));
```

Comment:
I think about something like return an error for specific member to the map to the value. Something like:
```
sponsor1Name -> password1
sponsor2Name -> ERROR: error when creating
sponsor3Name -> password3
sponsor4Name -> ERROR: error when creating
```
Then it would be on frondend to show these errors to user. What do you think @zlamalp ?
It is not standard behavior, need to be precisely defined in javadoc, but it should work much better than this.

Reply:
We discussed this further and come up with following solution.

You will technically return ``Map<String,Map<String,String>>``.
- Keys will be names of the sponsored members, so that GUI can map each result to the originating data.
- Value will be map again, with keys ``status, login and password``. Then in status you can put "OK" or "Error - ....message...." and in login/password you put generated UČO a password respectively.
  This way we can pass all necessary information to the GUI and extend it in the future if necessary.

Please add detailed description about object structure to the Javadoc.

Reply:
This looks better, thank you, I've changed the returned type based on this.